friends/src/App.js

```diff
@@ -74,17 +134,34 @@ class App extends Component {
           exact
           path="/friendslist"
           render={routeProps => (
-            <FriendsList {...routeProps} friends={this.state.friends} />
+            <FriendsList
+              {...routeProps}
+              friends={this.state.friends}
+              addFriend={this.addFriend}
+              editFriend={this.editFriend}
+              deleteFriend={this.deleteFriend}
+            />
+          )}
+        />
+        <Route
+          path="/addfriendform"
+          render={routeProps => (
+            <AddFriendForm
+              {...routeProps}
+              potentialFriendChanges={this.state.potentialFriendChanges}
+              handleChanges={this.handleChanges}
+              submitFriend={this.submitFriend}
+            />
           )}
         />
         <Route
-          path="/newfriendform"
+          path="/editfriendform"
           render={routeProps => (
-            <NewFriendForm
+            <EditFriendForm
               {...routeProps}
-              potentialNewFriend={this.state.potentialNewFriend}
+              potentialFriendChanges={this.state.potentialFriendChanges}
```

Comment:
maybe  next time you'll consider finding shorter names for your data, it looks like this was a pain to type out repetitively